Makefile

```diff
@@ -2,20 +2,25 @@
 
 CGO_ENABLED=0
 GOOS=linux
+CORE_IMAGES=./cmd/activator ./cmd/autoscaler ./cmd/controller ./cmd/queue ./cmd/webhook
+TEST_IMAGES=$(shell find ./test/test_images -mindepth 1 -maxdepth 1 -type d) ./test/controller
 
 install:
-	go install ./cmd/activator/ ./cmd/autoscaler/ ./cmd/controller/ ./cmd/queue/ ./cmd/webhook/
+	go install $(CORE_IMAGES)
 .PHONY: install
 
 test-install:
-	go build -o $(GOPATH)/bin/test-controller ./test/controller
-	go install ./test/test_images/autoscale ./test/test_images/envvars \
-	           ./test/test_images/helloworld ./test/test_images/httpproxy \
-	           ./test/test_images/pizzaplanetv1 ./test/test_images/pizzaplanetv2 \
-	           ./test/test_images/singlethreaded ./test/test_images/timeout \
-	           ./test/test_images/observed-concurrency
+	for img in $(TEST_IMAGES); do \
+		go install $$img ; \
+	done
 .PHONY: test-install
 
 test-e2e:
-	sh openshift/e2e-tests-openshift.sh
+	./openshift/e2e-tests-openshift.sh
 .PHONY: test-e2e
+
+# Generate Dockerfiles for core and test images used by ci-operator. The files need to be committed manually.
+generate-dockerfiles:
+	./openshift/ci-operator/generate-dockerfiles.sh openshift/ci-operator/knative-images $(CORE_IMAGES)
```

Comment:
I suppose you're talking about all the Dockerfiles in knative-images and knative-test-images folders. Those files were generated by calling `make generate-dockerfiles` manually and committed. The Dockerfiles have to exist in the source repository so that the ci-operator can pick them up.

Reply:
Yes, I get it now! And I read the comment as well! 🥇